### ISSUE TITLE
The Continual Learning Setting

### ISSUE THREAD

**Issue by appledora** [gyhandy/Channel-wise-Lightweight-Reprogramming#2]
Hi, first of all - thank you for writing such a well-thought out paper , with a great codebase. 

I was going through the training pipeline, and was curious about how exactly the inference would take place in a Continual learning setting? According to the paper, there will be **a pool of task-specific CLRs**. But in the code I can see that, _a separate model is trained for each task_. 
In that case, can you explain _how the task performance graphs were generated?_ Since the models are completely **isolated** from each other, isn't this more or less similar to normal fine-tuning on top of a large pretrained backbone? 

Apologies for the long question. I am just curious about how to utilize it for continual learning - i.e:, how it's actually tackling the `Catastrophic Forgetting` issue.

**Comment by Harry-Li-27** [comment]
Thank you for reaching out. So on line 73 in main.py, you will find that the optimizer only optimizes the CLR part of the model: "optimizer = optim.Adam(reg_params+noreg_params, lr=lr)". Also, in the model definition, take Res_Ghost() for example, all the parameters are fixed except for the CLR part of the model. So it means most part of the model is fixed and never got trained. Only the CLR part of the model, which in the code is Batchnorm, ReGhos_Block, and final fc layer. Please don't hesitate to ask me if anything remains unclear.

Best

**Comment by appledora** [comment]
ah...this makes sense. Is there any inference code available? 
Thanks again for taking the time to respond!

**Comment by Harry-Li-27** [comment]
The eval() function is doing the inference. If you are asking about doing inference for all models in the same time, we recommend you save the weight for the CLR block only through state_dict(), and then replace the weight for the CLR block for the new model.

**Comment by appledora** [comment]
Cool! I will try to implement your suggestions. Thanks!
